clang/lib/Basic/Targets/AMDGPU.h

```diff
@@ -111,6 +111,18 @@ class LLVM_LIBRARY_VISIBILITY AMDGPUTargetInfo final : public TargetInfo {
     return getPointerWidthV(AddrSpace);
   }
 
+  virtual bool isAddressSpaceSupersetOf(LangAS A, LangAS B) const override {
+    // The flat address space AS(0) is a superset of all the other address
+    // spaces used by the backend target.
```

Comment:
This should exclude several address spaces. Should exclude region. Arguably also the fat pointers

Reply:
The logic should only permit 0-5, I was torn on rejecting AS(2).

Reply:
Should reject it, it's not flat addressable. We don't have a magic aperture constant for it and the addrspacecast codegen will fail